src/test/java/reposense/util/TestUtil.java

```diff
@@ -154,16 +155,20 @@ public static boolean compareNumberFilesChanged(int expectedNumberFilesChanged,
         if (gitLogResult.isEmpty()) {
             return expectedNumberFilesChanged == 0;
         }
-        String[] changesLogged = gitLogResult.split("\n");
+        String[] changesLogged = gitLogResult.split(GitLog.COMMIT_INFO_DELIMITER);
         HashSet<String> filesChanged = new HashSet<>();
-        // Checks the 2nd line of each commit which contains info of the changed file
-        for (int i = 0; i < changesLogged.length; i += 4) {
-            String log = changesLogged[i + 1];
-            String fileChanged = log.split("\\| ")[0].trim();
-            if (fileChanged.contains("=>")) {
-                fileChanged = fileChanged.substring(fileChanged.indexOf("=> ") + 3);
+
+        // start from index 1 as index 0 is always empty.
+        for (int i = 1; i < changesLogged.length; i++) {
```

Comment:
If we have the same code for the production and test, there may be situations where they are both wrong. But I guess there are no better ways of checking it. Do be careful with these codes and make sure they are logically correct.

Reply:
WRT testing, I've verified that it works with all existing expected test values; and that the row where the number of files changed, number of lines modified are located at will always be consistent (located at the last line of each commit information).